app/src/sharedTest/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityTest.kt

```diff
@@ -104,8 +107,58 @@ private val SKILL_ID_LIST = listOf(FRACTIONS_SKILL_ID_0)
 @LooperMode(LooperMode.Mode.PAUSED)
 @Config(application = QuestionPlayerActivityTest.TestApplication::class, qualifiers = "port-xxhdpi")
 class QuestionPlayerActivityTest {
-  // TODO(#503): add tests for QuestionPlayerActivity (use StateFragmentTest for a reference).
+  // TODO(#503): Add more test-cases
   // TODO(#1273): add tests for Hints and Solution in Question Player.
+  /**
+   * Tests for QuestionPlayerActivityTest
+   *
+   * TOOLBAR
+   * []testQuestionPlayer_toolbarIsDisplayed
+   * []testQuestionPlayer_configChange_toolbarIsDisplayed
+   * [X]testQuestionPlayer_toolbarTitleIsDisplayed - match the string @string/question_player_title
+   * [X]testQuestionPlayer_configChange_toolbarTitleIsDisplayed
+   *
+   * BACK BUTTON
+   * [X]testQuestionPlayer_backPress_stopQuestionPlayerDialogIsDisplayed
+   * []testQuestionPlayer_backPress_clickCancel_dialogIsDismissed
+   * []testQuestionPlayer_backPress_clickLeave_questionPlayerIsClosed
+   *
+   * []testQuestionPlayer_onConfig_backPress_stopQuestionPlayerDialogIsDisplayed
+   * []testQuestionPlayer_onConfig_backPress_clickCancel_dialogIsDismissed
+   * []testQuestionPlayer_onConfig_backPress_clickLeave_questionPlayerIsClosed
+   *
+   * [X]testQuestionPlayer_backPress_configChange_stopQuestionPlayerDialogIsDisplayed
+   * []testQuestionPlayer_backPress_configChange_clickCancel_dialogIsDismissed
+   * []testQuestionPlayer_backPress_configChange_clickLeave_questionPlayerIsClosed
+   *
+   * []testQuestionPlayer_clickToolbarNav_stopQuestionPlayerDialogIsDisplayed
```

Comment:
@anandwana001 
shall I write the same test cases for `configChange` too?

Reply:
yes, you can start writing all

Reply:
After writing, when you send this pr for review, please add 2 screenshots in the PR description showing all the test cases are passing on espresso and robolectric. 

How to run a test on Robolectrci - https://github.com/oppia/oppia-android/wiki/Oppia-Android-Testing#running-robolectric-tests